### ISSUE TITLE
Problem with RTS5139

### ISSUE THREAD

**Issue by rafaelmaeuer** [0xFireWolf/RealtekCardReader#21]
Hey and thanks a lot for your kext.

I have problems getting the RTS5139 up and running on an ASUS UX32VD Laptop.
I got the info from a user that it should work with your kext, but I couldn't make it.

More details about the issue can be found on: https://github.com/rafaelmaeuer/Asus-UX32VD-Hackintosh/issues/36

**Comment by 0xFireWolf** [comment]
Oh, that's interesting. 
Your kernel log shows that the card status register reported that no card was inserted, even though you did insert a card.
If I remember it correctly, there are two ways to fetch the card status from the card reader, and it seems that the current approach adopted by this driver does not work on your machine.

**Comment by 0xFireWolf** [comment]
Please try the attached driver and add the boot argument `-rtsxppsta`. 
Check whether your card can be detected properly and upload the kernel log as per FAQ.

[RealtekCardReader_0.9.7_43b5887_DEBUG.zip](https://github.com/0xFireWolf/RealtekCardReader/files/7779311/RealtekCardReader_0.9.7_43b5887_DEBUG.zip)

**Comment by rafaelmaeuer** [comment]
I tried your v0.9.7 DEBUG kext with boot-arg, but the card reader isn't recognized:

![Bildschirmfoto 2022-01-07 um 11 32 16](https://user-images.githubusercontent.com/6418459/148531378-4ad69e0c-3259-46cb-9d11-e77b47b66177.png)

Here is the log file: [rtsx.log](https://github.com/0xFireWolf/RealtekCardReader/files/7828251/rtsx.log)

I have the Information of another user with the same machine whose card-reader works well, he sent me his EFI, but I couldn't get it to work...

I hope he doesn't mind if I share his image proof:
![image](https://user-images.githubusercontent.com/6418459/148532195-fdfdfdfb-0609-437d-8794-7ed43979f9ac.png)

**Comment by 0xFireWolf** [comment]
You kernel log still indicates that no card was inserted. 

> I have the Information of another user with the same machine whose card-reader works well, he sent me his EFI, but I couldn't get it to work...

Does that user need the boot argument `-rtsxppsta` to get the card reader working?
Is your card reader working properly under Ubuntu 20.04 LTS? You can boot the Live CD and check that.

**Comment by rafaelmaeuer** [comment]
Thanks for your reply I finally figured it out :)

Indeed I wasn't inserting a card when doing the previous log, as I thought it should show more details once working due to the helper kext. So I retried this and stumbled over following line:
```
System Policy: BDLDaemon(314) deny(1) file-read-metadata /private/var/folders/n1/2122st650c9c7f_5q__v7sxr0000gn/0/com.apple.nsurlsessiond/70D3879BCCBED0C0730B02BAF6FCDD03E3D14D00/sessions.plistSandbox: 2 duplicate reports for BDLDaemon
```

BDLDaemon ist the daemon of my bitdefender security system. I disabled Bitdefender Shield and voila:
![Bildschirmfoto 2022-01-09 um 15 10 30](https://user-images.githubusercontent.com/6418459/148686171-2e420b69-6439-441a-b435-252eb2cf4708.png)
It works as expected.

The only downside is I need to disable my Antivirus for SD-Cards to be recognized, but thats fine for now. Maybe I will figure out with Bitdefender how to add an exception for this...

I also tried linux-boot-stick with ubuntu 20.04 without any success of the card been recognized and I did also check if the boot-arg is required. It does work without boot-arg and kext version 0.9.6.

Thank you very much for your help.
